### PR TITLE
Remove use of dictionary for output. Can't figure out how to generate…

### PR DESCRIPTION
**Pull request by jjtan**
… a single metric for hyper param tuning with the dict approach. Will try this method out.

Issue #42 

Pro: A single output seems to be better supported and more common.
Cons: We lost the metrics per output during training (for now).

auc_tb

### CODE REVIEW

File: kaggle-classification/keras_trainer/cnn_with_attention.py [hunk 1]
```diff
@@ -62,14 +62,9 @@ def get_model(self):
     X = Dropout(self.hparams.dropout_rate)(X)
     X = Dense(128, activation='relu')(X)
     X = Dropout(self.hparams.dropout_rate)(X)
-    toxic_out = Dense(1, activation='sigmoid', name='toxic')(X)
-    severe_toxic_out = Dense(1, activation='sigmoid', name='severe_toxic')(X)
-    obscene_out = Dense(1, activation='sigmoid', name='obscene')(X)
-    threat_out = Dense(1, activation='sigmoid', name='threat')(X)
-    insult_out = Dense(1, activation='sigmoid', name='insult')(X)
-    identity_hate_out = Dense(1, activation='sigmoid', name='identity_hate')(X)
+    O = Dense(6, activation='sigmoid', name='outputs')(X)
```
[the remark this answers is not in the document]
Comment (ericagreene) [reply kaggle-classification/keras_trainer/cnn_with_attention.py]: Since you're not naming the different outputs anymore, how do you tell which output is associated with each subtype?

File: kaggle-classification/keras_trainer/model.py [hunk 2]
```diff
@@ -42,11 +42,11 @@
     learning_rate=0.00005,
     dropout_rate=0.5,
     batch_size=128,
-    epochs=10,
+    epochs=1,
     sequence_length=250,
     embedding_dim=100,
     train_embedding=False,
-    model_type='single_layer_cnn',
+    model_type='cnn_with_attention',
```
Comment (ericagreene) [inline kaggle-classification/keras_trainer/model.py]: I think we should leave the basic model as the default unless there's a good reason to do otherwise.

File: kaggle-classification/keras_trainer/model.py [hunk 3]
```diff
@@ -115,13 +115,11 @@ def predict(self, texts):
...
   def score_auc(self, data):
     predictions = self.predict(data['comment_text'])
-    scores = []
-    for idx, label in enumerate(LABELS):
-      labels = np.array(data[label])
-      score = metrics.roc_auc_score(labels, predictions[idx].flatten())
-      scores.append(score)
-      print('{} has AUC {}'.format(label, score))
-    print('Avg AUC {}'.format(np.mean(scores)))
+    labels = np.array(list(zip(*[data[label] for label in LABELS])))
```
Comment (ericagreene) [inline kaggle-classification/keras_trainer/model.py]: Whenever you do some fancy list comprehension magic, it's good to add a comment to explain to the uninitiated what the hell is going on :)

File: kaggle-classification/keras_trainer/model.py [hunk 4]
```diff
@@ -115,13 +115,11 @@ def predict(self, texts):
...
+    individual_auc_scores = metrics.roc_auc_score(labels, predictions, average=None)
+    print("Individual AUCs: {}".format(list(zip(LABELS, individual_auc_scores))))
+    mean_auc_score = metrics.roc_auc_score(labels, predictions, average='macro')
+    print("Mean AUC: {}".format(mean_auc_score))
```
Comment (ericagreene) [inline kaggle-classification/keras_trainer/model.py]: Don't change it in this PR, but we should switch from `print` to using the TensorFlow logging library, `tf.logging()`

File: kaggle-classification/bin/run_keras_local.sh [hunk 5]
```diff
@@ -21,4 +13,12 @@ echo ""
...
\ No newline at end of file
+echo ""
+
+python -m keras_trainer.model \
+	--train_path=${INPUT_PATH}/train.csv \
+	--validation_path=${INPUT_PATH}/validation.csv \
+	--embeddings_path=${INPUT_PATH}/glove.6B/glove.6B.100d.txt \
+	--job-dir=${OUTPUT_PATH} \
+	--log_path=${LOG_PATH} \
+	--hparams=epochs=1
```
[the remark this answers is not in the document]
Comment (jjtan) [reply kaggle-classification/bin/run_keras_local.sh]: hparams is actually a dict so I think you'd have to do it this way. I did remove hparams in the follow up CL when I implement hyper param tuning - will need to speak with @nthain about it to see if we are giving up any advantages from using it.

File: kaggle-classification/keras_trainer/cnn_with_attention.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jjtan) [reply kaggle-classification/keras_trainer/cnn_with_attention.py]: There is a constant with all the labels - there is an ordering to them. Not ideal but couldn't figure out a way to preserve the dictionary and do hyper param tuning too
